mm2src/mm2_bitcoin/chain/src/block_header.rs

```diff
@@ -777,6 +802,12 @@ mod tests {
         assert_eq!(serialized.take(), headers_bytes);
     }
 
+    #[test]
+    fn test_firo_block_headers_prog_pow() {
```

Comment:
Similar to other tests, please check that header is then serialized to the initial input.